Userland/Libraries/LibGfx/ICOLoader.cpp

```diff
@@ -123,7 +123,7 @@ static size_t find_largest_image(ICOLoadingContext const& context)
     size_t index = 0;
     size_t largest_index = 0;
     for (auto const& desc : context.images) {
-        if (desc.width * desc.height > max_area) {
+        if (static_cast<size_t>(desc.width) * static_cast<size_t>(desc.height) > max_area) {
```

Comment:
How about adding a `VERIFY` here that these two are non-negative?

Reply:
I think this might not be necessary, since these values are `u16`s, I think. They are defined here:

https://github.com/SerenityOS/serenity/blob/5c7255f0a74ae826fc955875ba26a81c60787def/Userland/Libraries/LibGfx/ICOLoader.cpp#L70-L76

Reply:
In that case, what this does change achieve? If both width and height are already unsigned (and since max_area is unsigned), this is already an unsigned comparison

Reply:
It does look like it, however I get an signed comparison warning (error, even) nonetheless. I believe that might be happening because arithmetic operators do not accept types smaller than int, so the product of two `u16`s gets promoted to `int`:

https://en.cppreference.com/w/cpp/language/implicit_conversion

This change gets rid of that warning.

Reply:
> I get an signed comparison warning (error, even)

Not sure which warning/error you're referring to, CI passes fine.

In which case, I'm not sure this fixes anything, as the comparison casts said promoted int back to an unsigned value, so it results in the expected behaviour.

Reply:
> Not sure which warning/error you're referring to, CI passes fine.

Yeah, I figured it was something weird with my setup, that's why I didn't open an issue about that.

Anyhow, I still thought it was worth "fixing", since it was simple enough and the code with the change seemed a little more correct. I'm clearly out of my depth regarding C++ specifics, and I'm specially new to this codebase, so I'm likely wrong about this, but I thought there were some (unlikely) inputs that code break the original code. For example, the program

```c++
#include <unistd.h>
#include <stdio.h>

int main()
{
        unsigned short width = 0xFFFF;
        unsigned short height = 0xFFFF;
        size_t area = 0xFFFFFFFF;

        if (width * height > area) {
                printf("1\n");
        }

        if (static_cast<size_t>(width) * static_cast<size_t>(height) > area) {
                printf("2\n");
        }

        return 0;
}
```

prints `1`, but not `2`. To me, that indicates there is some difference in behaviour, with the second being "more correct". Anyway, I still might be missing something. In that case, feel free to close this. Thanks! :^)

Reply:
I see, the promotion to int results in different multiplication overflow behaviour, so indeed, this improved the situation. My only remaining gripe is with that of the commit title, from the contribution guidelines:
> **Don't:**
> - Use weasel-words like "refactor" or "fix" to avoid explaining what's being changed.

What did you change?

Reply:
Ah, my bad. Changed the commit message. How does this look?